File: fr.sivrit.svn.helper.java/src/fr/sivrit/svn/helper/java/svn/SvnNode.java
package fr.sivrit.svn.helper.java.svn;

import java.util.Collection;
import java.util.Collections;

import org.tigris.subversion.svnclientadapter.SVNUrl;

public final class SvnNode {
    public final SVNUrl url;
    public final String content;
    public final Collection<SvnFolderEntry> children;
    public final boolean isDir;
    public final long version;

    public SvnNode(final SVNUrl url, final long version, final Collection<SvnFolderEntry> children) {
        super();
        this.url = url;
        this.content = null;
        this.children = children;
        this.isDir = true;
        this.version = version;
    }

    public SvnNode(final SVNUrl url, final long version, final String content) {
        super();
        this.url = url;
        this.content = content;
        this.children = Collections.emptySet();
        this.isDir = false;
        this.version = version;
    }
}

File: fr.sivrit.svn.helper.scala/src/fr/sivrit/svn/helper/scala/svn/SvnNode.scala
package fr.sivrit.svn.helper.scala.svn
import org.tigris.subversion.svnclientadapter.SVNUrl
import org.tigris.subversion.svnclientadapter.ISVNDirEntry
import org.tigris.subversion.svnclientadapter.SVNNodeKind

case class SvnFolderEntry(name: String, version: Long, isDir: Boolean) {
  def this(entry: ISVNDirEntry) = this(
    entry.getPath,
    entry.getLastChangedRevision.getNumber,
    entry.getNodeKind == SVNNodeKind.DIR)
}

sealed abstract class SvnNode {
  def url: SVNUrl
  def version: Long
}
case class SvnFile(url: SVNUrl, version: Long, content: String) extends SvnNode
case class SvnDir(url: SVNUrl, version: Long, children: Set[SvnFolderEntry]) extends SvnNode

